### PR TITLE
doc: improve main tock docs

### PR DESCRIPTION
**Pull request by bradjc**
Generally make docs more consistent and update to latest code.

### CODE REVIEW

File: doc/Compilation.md [hunk 1]
```diff
@@ -308,33 +320,33 @@ architectures.
...
 In addition, you must specify the sources for your application:
 
-  - `C_SRCS` - A list of C files to compile
-  - `CXX_SRCS` - A list of C++ files to compile
-  - `AS_SRCS` - A list of assembly files to compile
-  - `EXTERN_LIBS` - A list of directories for libraries [**compiled for Tock**](#compiling-libraries-for-tock)
+  - `C_SRCS`: A list of C files to compile.
+  - `CXX_SRCS`: A list of C++ files to compile.
+  - `AS_SRCS`: A list of assembly files to compile.
+  - `EXTERN_LIBS`: A list of directories for libraries [**compiled for Tock**](#compiling-libraries-for-tock).
```
Comment (ppannuto) [inline doc/Compilation.md]: What's the motivation for the change from `:` to `-`? I feel like the `:` attached to the padded teletype looks weird:

image

File: doc/Compilation.md [hunk 2]
```diff
@@ -366,8 +378,8 @@ libexample/                <-- Folder name must match library name
 Like applications, libraries can leverage the Tock build system to do most of
 the heavy lifting. Simply,
 
-  1. Set `TOCK_USERLAND_BASE_DIR` to the path to the Tock userland
-  2. `include $(TOCK_USERLAND_BASE_DIR)/TockLibrary.mk`
+  1. Set `TOCK_USERLAND_BASE_DIR` to the path to the Tock userland.
+  2. `include $(TOCK_USERLAND_BASE_DIR)/TockLibrary.mk`.
```
Comment (ppannuto) [inline doc/Compilation.md]: I feel like the period is a disservice here, looks funny and inhibits copy-paste

File: doc/Syscalls.md [hunk 3]
```diff
@@ -94,14 +112,15 @@ The Yield syscall takes no arguments.
...
+Subscribe assigns callback functions to be executed in response to various
+events.
 
 The Subscribe syscall takes two arguments:
 
  - `subscribe_number`: An integer index for which function is being subscribed
  - `callback`: A pointer to a callback function to be executed when this event
  occurs. All callbacks conform to the C-style function signature:
- `void callback(int arg1, int arg2, int arg3, void* data)`
+ `void callback(int arg1, int arg2, int arg3, void* data)`.
```
Comment (ppannuto) [inline doc/Syscalls.md]: Again think this period is weird, particularly b/c this isn't a sentence.

If you're really attached to the period, you should probably also add one to the line above (though I'd prefer neither)

File: doc/Compilation.md (hunk 1 above)
Comment (bradjc) [inline doc/Compilation.md]: Consistency. I also find it to look better in the raw source.

File: doc/Syscalls.md (hunk 3 above)
Comment (bradjc) [inline doc/Syscalls.md]: Ah I did miss one. But I do contend that "All callbacks conform..." is a sentence.
